### PR TITLE
Password Change

### PR DESCRIPTION
**Pull request by djschilling**
Implement the first part of #27.

### CODE REVIEW

File: lib/radiator_web/controllers/admin/usersettings_controller.ex [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  alias Radiator.Auth
+
+  action_fallback RadiatorWeb.Api.FallbackController
+
+  def index(conn, _params) do
+    render(conn, "usersettings.html", changeset: Auth.Register.change_user(%Auth.User{}))
+  end
+
+  def create(conn, params) do
+    user_map = params["user"]
```
Comment (eteubert) [inline lib/radiator_web/controllers/admin/usersettings_controller.ex]: `user_params` is the more common variable name choice here. Variable type is not usually included in the name.

File: lib/radiator_web/controllers/admin/usersettings_controller.ex [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    cond do
```
Comment (eteubert) [inline lib/radiator_web/controllers/admin/usersettings_controller.ex]: If all conditions of cond are true and false, use if/then/else.

File: lib/radiator_web/controllers/admin/usersettings_controller.ex [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  alias Radiator.Auth
+
+  action_fallback RadiatorWeb.Api.FallbackController
+
+  def index(conn, _params) do
+    render(conn, "usersettings.html", changeset: Auth.Register.change_user(%Auth.User{}))
+  end
+
+  def create(conn, params) do
```
Comment (eteubert) [inline lib/radiator_web/controllers/admin/usersettings_controller.ex]: Maybe `update`? You're not creating anything.

Comment (djschilling) [reply lib/radiator_web/controllers/admin/usersettings_controller.ex]: I saw that the `resources` function was used in the admin routes. And there i think the create method which stands for `POST /admin/usersettings` is the best match.

Should i switch the routes to not use the `resources` function, and define the http methods directly?

Comment (eteubert) [reply lib/radiator_web/controllers/admin/usersettings_controller.ex]: You probably need more flexibility here than the `resources` macro provides. It's ok to define custom routes and ideal http methods.

**Comment by djschilling** [comment]
All mentioned things are fixed

**Comment by eteubert** [comment]
One way to simplify the controller update logic is `with`, which lets you focus on the happy path first and deal with errors later. Something like this:

```elixir
    with {:ok, _} <- User.check_password(user, user_params["password_current"]),
         {:ok, _user} <- Auth.Register.update_user(user, user_params) do
      conn
      |> put_flash(:info, "Successfully updated password.")
      |> redirect(to: Routes.admin_network_path(conn, :index))
    else
      # handle error cases here
    end
```

Writing a custom Ecto changeset function may also help increase readability.

**Comment by djschilling** [comment]
That looks nice.

The implementation now uses a seperate changeset and the `with` keyword.

File: lib/radiator/auth/user.ex [hunk 4]
```diff
@@ -45,11 +46,34 @@ defmodule Radiator.Auth.User do
     |> validate_format(:name, ~r/^[^\s ]+$/)
     |> validate_format(:email, ~r/^\S+@\S+$/)
     |> validate_length(:name, min: 2, max: 99)
+    |> validate_length(:password, min: 2)
     |> validate_required([:email, :name])
     |> encrypt_password
     |> validate_required([:password_hash])
   end
 
+  def change_password_changelog(%User{} = user, attrs) do
```
Comment (eteubert) [inline lib/radiator/auth/user.ex]: typo: changeset, not changelog

**Comment by eteubert** [comment]
Aside from the typo, looks good now 👍

**Comment by djschilling** [comment]
Tipo fixed.

**Pull request merged**